db/db_impl_open.cc

```diff
@@ -383,11 +383,22 @@ Status DBImpl::Recover(
   }
 
   Status s = versions_->Recover(column_families, read_only);
+
+  // create persistent_stats CF for the first time
+  // DB mutex is already held
+  if (mutable_db_options_.persist_stats_to_disk) {
+    s = CreatePersistStatsHandle();
+  }
   if (immutable_db_options_.paranoid_checks && s.ok()) {
     s = CheckConsistency();
   }
   if (s.ok() && !read_only) {
     for (auto cfd : *versions_->GetColumnFamilySet()) {
+      // skip directory creation if persistent_stats CF was just created
```

Comment:
because it's already created in https://github.com/facebook/rocksdb/blob/master/db/db_impl.cc#L1972 as part of the call to `CreatePersistStatsHandle()->CreateColumnFamily()`

Reply:
there are two code paths when it comes to creating/recovering persistent stats column family:
* creating for the first time: `InitPersistStatsColumnFamily` calls `DBImpl::CreateColumnFamilyImpl`  which add directories (see https://fburl.com/azkputvf) for newly created column family
* recovering: `InitPersistStatsColumnFamily` just creates cf handle and no directories is added
We need to skip re-creating directories for the first case which is why we handle it differently

Reply:
It seems to be wrong that we first do 2 and then 1. If we can recover stats CF, we should recover from there. Only if itr is not there, we should create a new one.

Reply:
Makes sense, will move the call `InitPersistStatsColumnFamily` to after creating directories.